aws_lambda_powertools/metrics/metrics.py

```diff
@@ -122,10 +131,13 @@ def decorate(event, context):
                 if capture_cold_start_metric:
                     self.__add_cold_start_metric(context=context)
             finally:
-                metrics = self.serialize_metric_set()
-                self.clear_metrics()
-                logger.debug("Publishing metrics", {"metrics": metrics})
-                print(json.dumps(metrics))
+                if not raise_for_empty_metrics and not self.metric_set:
```

Comment:
Can we move this to the domain level code (`metrics/base.py`)? That way, we can 

1) quickly implement to `single_metric` if we need to too
2) accommodate those flushing metrics manually like Richard in another issue
3) consider whether this should also be an env var, and possibly in the constructor if point 1 is requested by customers

Reply:
I added it to the decorator rather than base because I think the behavior should be different for the decorator. If I'm explicitly making a call to serialize metrics when their are none, throwing an exception is fine and I can handle it myself.